repository/src/main/resources/alfresco/ibatis/org.alfresco.repo.domain.dialect.Dialect/node-common-SqlMap.xml

```diff
@@ -1147,6 +1147,20 @@
             parentNode.id = #{parentNode.id} and
             a.child_node_id IS NULL
     </select>
+
+    <select id="select_LocalNameListOfChildAssocsWithoutParent" parameterType="ChildAssoc" resultType="java.lang.String">
+        select distinct c.qname_localname
+        from alf_child_assoc c
+        where c.parent_node_id = #{parentNode.id}
+        and not exists (
+            select 1
+            from alf_child_assoc a
+            join alf_child_assoc z
+            on z.parent_node_id = #{parentNode.id}
+            and z.child_node_id = a.parent_node_id
+            where c.child_node_id = a.child_node_id

```

Comment:
Going through this query then it will pick out E as a "child assoc without parent":
```
A
|\
B \
C  |
D /
|/
E
```
I don't think this is what you wanted.

Reply:
We discussed this and it turns out to be a limitation with the original query. Consequently there's no need to fix it, we can just name the method appropriately.